magefile.go

```diff
@@ -1154,15 +1216,319 @@ func packageAgent(platforms []string, packagingFn func()) {
+			log.Printf("checking if %s:%v should be included", pkgName, pkgDesc)
+		}
+		for _, pkgOS := range pkgDesc.Os {
+			platformString, _ := mapManifestPlatformToAgentPlatform(fmt.Sprintf("%s/%s", pkgOS, pkgDesc.Architecture))
+			if slices.Contains(platforms, platformString) {
+				if mg.Verbose() {
+					log.Printf("platforms include %s", platformString)
+				}
+				filteredPackages[pkgName] = pkgDesc
+				break
+			}
+		}
+	}
+	if mg.Verbose() {
+		log.Printf("filtered packages: %v", filteredPackages)
+	}
+	return filteredPackages
+}
+
+func downloadDRAArtifacts(ctx context.Context, manifestUrl string, downloadDir string, projects ...string) (map[string]tools.Package, error) {
+
+	build, err := manifest.DownloadManifest(manifestUrl)
+	if err != nil {
+		return nil, fmt.Errorf("downloading manifest from %q: %w", manifestUrl, err)
+	}
+
+	// Create a dir with the buildID at <downloadDir>/<buildID>
+	draDownloadDir := filepath.Join(downloadDir, build.BuildID)
+	err = os.MkdirAll(draDownloadDir, 0o770)
+
+	if err != nil {
+		return nil, fmt.Errorf("creating %q directory: %w", err)
+	}
+
+	if mg.Verbose() {
+		log.Printf("url:%q, err %v", manifestUrl, err)
+	}
+
+	// sync access to the downloadedArtifacts map
+	mx := new(sync.Mutex)
+	downloadedArtifacts := map[string]tools.Package{}
+	errGrp, errCtx := errgroup.WithContext(ctx)
+
+	for _, projectName := range projects {
+		project, ok := build.Projects[projectName]
+		if !ok {
+			return nil, fmt.Errorf("project %q not found in manifest at %q", projectName, manifestUrl)
+		}
+
+		if mg.Verbose() {
+			log.Printf("build %q project %s packages: %+v", build.BuildID, projectName, project)
+		}
+		// filter down the packages to the platforms we are building/support
+		filteredPackages := filterPackagesByPlatform(project.Packages)
+		if mg.Verbose() {
+			log.Printf("packages to download: %v", filteredPackages)
+		}
+		for pkgName, pkgDesc := range filteredPackages {
+			downloadFunc := func(pkgName string, pkgDesc tools.Package) func() error {
+				return func() error {
+					artifactDownloadPath := filepath.Join(draDownloadDir, pkgName)
+					err := manifest.DownloadPackage(errCtx, pkgDesc.URL, artifactDownloadPath)
+
+					if err != nil {
+						return fmt.Errorf("downloading %q: %w", pkgName, err)
+					}
+
+					// download the SHA to check integrity
+					artifactSHADownloadPath := filepath.Join(draDownloadDir, pkgName+".sha512")
+					err = manifest.DownloadPackage(errCtx, pkgDesc.ShaURL, artifactSHADownloadPath)
+					if err != nil {
+						return fmt.Errorf("downloading SHA for %q: %w", pkgName, err)
+					}
+
+					err = download.VerifyChecksum(sha512.New(), artifactDownloadPath, artifactSHADownloadPath)
+					if err != nil {
+						return fmt.Errorf("validating checksum for %q: %w", pkgName, err)
+					}
+
+					// we should probably validate the signature, it can be done later as we return the package metadata
+					// see https://github.com/elastic/elastic-agent/issues/4445
+
```

Comment:
[Suggestion]
Then I think it'd be worth adding a warning log so it's clear the signature isn't verified

Reply:
The signature is never verified anywhere in the packaging process.

This enhancement is already tracked by the issue in the comment, a warning would just appear multiple times in the logs of the early releases pipelines, it's not for any final user just for packagers. I would prefer to add a debug log saying that we verified the signature when the check is implemented